middlewares/validators/external-accounts.js

```diff
@@ -1,11 +1,19 @@
 const joi = require("joi");
 
 const externalAccountData = async (req, res, next) => {
-  const schema = joi.object().strict().keys({
-    type: joi.string().required(),
-    token: joi.string().required(),
-    attributes: joi.object().strict().required(),
-  });
+  const schema = joi
+    .object()
+    .strict()
+    .keys({
+      type: joi.string().required(),
+      token: joi.string().required(),
+      attributes: joi.object().required().keys({
+        discordId: joi.number().optional(),
+        username: joi.string().optional(),
+        picture: joi.string().optional(),
+        expiry: joi.number().optional(),
+      }),
```

Comment:
just wanted to confirm that since all individual keys are optional. does attributes need to be an empty object if none of the child keys are there.

Reply:
No attributes can't be empty object as it will contain the data of external accounts, and child keys will vary according to the type of external account

Reply:
- I needed a suggestion on naming the `discordId` key, should we name it `discordId` or `id`. 
- The usecase of this API is generic and can be used for any external service or account. 
- But on the other hand, user of this API may get confused with some other id while using it for discord bots

Reply:
I remember @RitikJaiswal75 having a discussion with Ankush on the same , I think Ankush said we can call it `socialId` or `thirdPartyId` or something on similar lines. @RitikJaiswal75 what do you think.

Reply:
Thanks, will confirm it

Reply:
> No attributes can't be empty object as it will contain the data of external accounts, and child keys will vary according to the type of external account

In this case we can add a validation that at least 1 child key is required . something like 
```
.keys({
      discordId: joi.number().optional(),
      username: joi.string().optional(),
      picture: joi.string().optional(),
      expiry: joi.number().optional(),
}).min(1)
```

Reply:
Made the change, thanks for the suggestion